### PR TITLE
feat: Create distribution.GroupNode (improve pkg interface)

### PR DESCRIPTION
**Pull request by michaljurecko**
Jira: https://keboola.atlassian.net/browse/PSGO-416

**Changes:**
- Created `distribution.Node.Group(<name>)` method.

-----------

Before: 
- You had to create `distribution.Node` multiple times, for each distribution group.
- Each group had a separate getter in the `dependencies`.

After:
- Now, there is only the one dependency `DistributionNode() *distribution.Node`.
- New group can be created in a application part using `distribution.Node.Group(<name>)`.
- This simplifies dependencies.

-----

Package docs:
https://github.com/keboola/keboola-as-code/blob/b382721ea4a1ad99a2cc8ac704be01cd93e651fe/internal/pkg/service/common/distribution/doc.go#L1-L69

### CODE REVIEW

**Review by hosekpeter — APPROVED** [review]
LGTM

File: internal/pkg/service/common/distribution/listener.go [hunk 1]
```diff
@@ -31,20 +29,19 @@ type listeners struct {
...
 	v := &listeners{
-		config:    n.config,
+		config:    cfg,
 		lock:      &sync.Mutex{},
 		listeners: make(map[listenerID]*Listener),
 	}
 
 	// Graceful shutdown
-	ctx, cancel := context.WithCancel(context.Background()) // nolint: contextcheck
+	ctx, cancel := context.WithCancel(ctx)
```
[the remark this answers is not in the document]
Comment (michaljurecko) [reply internal/pkg/service/common/distribution/listener.go:42]: Thanks, I'll check it out.

**Comment by michaljurecko** [comment]
(sorry for missing comments)

File: internal/pkg/service/common/distribution/listener.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (michaljurecko) [reply internal/pkg/service/common/distribution/listener.go]: It is valid, but can be simplified, I will explain it in more detail at the standup.

https://github.com/keboola/keboola-as-code/pull/1588/commits/7bbd13ebaf2ee198ff873ca2bc166d05ae25effb

**Pull request merged**